### PR TITLE
Added fallback method select container by name

### PR DESCRIPTION
**Pull request by TTmaister**
Proposal #344 

Fallback method if container id has not provided.

Read ContainerName environment variable and find first container by name

### CODE REVIEW

**Comment by TTmaister** [comment]
@microsoft-github-policy-service agree

**Review by xiaomi7732 — CHANGES_REQUESTED** [review]
Thanks for the contribution. It largely looks good to me. There's only one small piece about null handling, do you mind taking a look? (y)

File: src/ApplicationInsights.Kubernetes/Containers/ContainerIdHolder.cs [hunk 1]
```diff
@@ -44,12 +45,26 @@ public bool TryBackFillContainerId(V1Pod pod, out V1ContainerStatus? containerSt
...
+            if (containerStatuses.Count == 1)
+            {
+                containerStatus = containerStatuses[0];
+                _logger.LogDebug(FormattableString.Invariant($"Use the only container inside the pod for container id: {containerStatus.ContainerID}"));
+            }
+            else
+            {
+                string? containerName = Environment.GetEnvironmentVariable("ContainerName");
+                _logger.LogDebug(FormattableString.Invariant($"Select container by environment variable containerName: {containerName}"));
+                containerStatus = containerStatuses.FirstOrDefault(c => c.Name == containerName);
```
[the remark this answers is not in the document]
Comment (TTmaister) [reply src/ApplicationInsights.Kubernetes/Containers/ContainerIdHolder.cs]: Changed to string equals.

[the remark this answers is not in the document]
Comment (TTmaister) [reply src/ApplicationInsights.Kubernetes/Containers/ContainerIdHolder.cs]: Added check that `containerStatus` is not null.

File: src/ApplicationInsights.Kubernetes/Containers/ContainerIdHolder.cs [hunk 2]
```diff
@@ -44,12 +45,26 @@ public bool TryBackFillContainerId(V1Pod pod, out V1ContainerStatus? containerSt
...
+                _logger.LogDebug(FormattableString.Invariant($"Selected container by container.name property container id: {containerStatus.ContainerID}"));
+            }
+
+            _logger.LogInformation(FormattableString.Invariant($"Selected container {containerStatus.Name} container id: {containerStatus.ContainerID}"));
 
             using (IServiceScope scope = _serviceScopeFactory.CreateScope())
```
[the remark this answers is not in the document]
Comment (TTmaister) [reply src/ApplicationInsights.Kubernetes/Containers/ContainerIdHolder.cs]: Also check that `containerStatus` is not null.

**Review by xiaomi7732 — APPROVED** [review]

**Pull request merged**